pkg/generator/generator.go

```diff
@@ -118,8 +143,108 @@ func NewGenerator(apiVersion, kind, projectName, repoPath string) *Generator {
 // │   |   ├── build
 // │   |   └── codegen
 // │   └── version
+//
+// Render generates the following project structure for Operator type Ansible
+// ├── <projectName>
+// │   ├── Dockerfile
+// │   ├── roles
+// │   │   └── <kind>
+// │   ├── watches.yaml
+// │   ├── deploy
+// │   │   ├── <kind>-CRD.yaml
+// │   │   ├── rbac.yaml
+// │   │   ├── operator.yaml
+// │   ├   └── cr.yaml
+
 func (g *Generator) Render() error {
-	if err := g.generateDirStructure(); err != nil {
+	switch g.operatorType {
+	case ansibleOperatorType:
+		if err := g.renderAnsibleOperator(); err != nil {
+			return err
+		}
+	case goOperatorType:
+		if err := g.renderGoOperator(); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unexpected operator type [%v]", g.operatorType)
+	}
+	return nil
+}
+
+func (g *Generator) renderAnsibleOperator() error {
+	if err := g.generateAnsibleDirStructure(); err != nil {
+		return err
+	}
+	if err := g.renderTmp(); err != nil {
+		return err
+	}
+	if err := g.renderRole(); err != nil {
+		return err
+	}
+	if err := g.renderWatches(); err != nil {
+		return err
+	}
+	if err := g.renderPlaybook(); err != nil {
+		return err
+	}
+	return g.renderDeploy()
+}
+
+func (g *Generator) renderRole() error {
+	fmt.Printf("Rendering Ansible Galaxy role [%v/%v/%v]...\n", g.projectName, rolesDir, g.kind)
+	agcmd := exec.Command(filepath.Join(g.projectName, galaxyInitCmd))
+	_, err := agcmd.CombinedOutput()
```

Comment:
It might be worth logging or returning the CombinedOutput here as well if the command fails.
The galaxyInit script might run but exit with an error. In which case the stdout and stderr might be more useful to see.
The err returned will just only say that the command exited with exit code 1.